dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/DynamoDbEnhancedBuildTimeConfig.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.amazon.dynamodb.enhanced.runtime;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigDocSection;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "dynamodbenhanced", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class DynamoDbEnhancedBuildTimeConfig {
+
+    /**
+     * Extension of the DynamodbEnhancedClient
+     */
+    @ConfigItem
+    @ConfigDocSection
+    Optional<List<DynamodbEnhancedClientExtension>> clientExtensions = Optional
+            .of(List.of(DynamodbEnhancedClientExtension.VersionedRecordExtension,
+                    DynamodbEnhancedClientExtension.AutoGeneratedTimestampRecordExtension));
+
+    public enum DynamodbEnhancedClientExtension {
+        VersionedRecordExtension,
+        AtomicCounterExtension,
+        AutoGeneratedTimestampRecordExtension
+
+    }
```

Comment:
Ok, I understand your thoughts. 

I think per config we should only support extensions provided by aws-sdk.
 
For custom extensions a simple producer would work? 

Or do you have a idea, how we could support custom extensions also?

Reply:
To avoid the closed enum, maybe support extensions like quarkus cxf support features in application.properties ?

https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/dev/user-guide/cxf-interceptors-and-features.html
https://github.com/quarkiverse/quarkus-cxf/blob/b23ec118730538d2f6532e1df786de39d684f2cd/extensions/core/runtime/src/main/java/io/quarkiverse/cxf/transport/CxfHandler.java#L120

It first try to load a named bean and if not available it instanciates the class.

The configuration would look like

````
quarkus.dynamodbenhanced.client-extensions=software.amazon.awssdk.enhanced.dynamodb.extensions.VersionedRecordExtension,MyCustomNamedBeanProvideByAProducer
````

Reply:
you may also look at how interceptors are handled here

Reply:
I think it is probably easier to support the extensions provides by the sdk at first. But I really like the idea from @scrocquesel to config this via properties which also enables custom extensions. If the effort is too high for the beginning we should stick to the provided extensions.

Reply:
I played a bit to find something working https://github.com/knuspertante/quarkus-amazon-services/pull/1.
At least, I would keep the fully qualified name for the config.

I also add the generation of the documentation of the config items.

Reply:
@scrocquesel thx for your work! I merged your [PR#1](https://github.com/knuspertante/quarkus-amazon-services/pull/1)

Reply:
Could you check that it works in native ? If not, we may need to pass a collection of `RecorderContext recorderContext.classProxy(className)` to the recorder instead of the set of class names (or produce `ReflectiveClassBuildItem`). RecorderContext can be injected in the build step. See https://quarkus.io/guides/writing-extensions#service-files

Reply:
Sure. I run the tests in native-mode. So far, all works great ;-) Thanks for your work!

Reply:
I think we should go with @scrocquesel suggestion and provide shortcurts for well know classes.

You could have `versioned-record`, `atomic-counter` and `auto-generated-timestamp-record` as shortcuts that would be automatically transformed to the SDK classes and you could also push your own classes.

That way you could have the best of both worlds. The shortcuts would be in the javadoc so properly documented.

Reply:
I push a fix to @knuspertante repo. This was not working for custom extension.

@gsmet This is not what have been done for interceptors, and we will need to support the mapping on the long term.

From the integration tests, I understood that such extensions works with annotations on table entity. Maybe we can plan to discover needed extensions and add them from the model if the property is not set ?

Reply:
> @gsmet This is not what have been done for interceptors, and we will need to support the mapping on the long term.

For the former, I don't know. For the latter, probably better that we maintain the mapping centrally rather than requiring people to update their applications if they change.
But I won't fight on this, feel free to ignore me :).

Reply:
My personal opinion: the current status is the most consistent way, since only extensions of the aws-sdk could be currently recognized via annotations. Also all extensions are specified with full classnames.